ipam/allocator.go

```diff
@@ -473,8 +473,11 @@ func (alloc *Allocator) encode() []byte {
 	}
 
 	// We're only interested in Paxos until we have a Ring.
+	// Non-electing participants (e.g. observers) return
+	// a nil gossip state in order to provoke a unicast ring
+	// update from electing peers who have reached consensus.
```

Comment:
It looks to me that an allocation attempt will call `establishRing()` and will lead to a gossip broadcast, which in the case of an observer will have the nil-nil state that triggers an election in the electorate.